tests/clone.rs

```diff
@@ -36,6 +36,27 @@ fn clone_and_references() {
     assert_eq!(closure(), ());
 }
 
+#[test]
+fn subfields_renaming() {
+    struct Foo {
+        v: Rc<usize>,
+    }
+
+    impl Foo {
+        fn foo(&self) {
+            let state = Rc::new(RefCell::new(State::new()));
+
+            let closure = clone!(@strong self.v as v, @weak state as hello => move |_| {
```

Comment:
Can you also add a test for `self` renaming?

Reply:
Already done in the docs.

Reply:
Ok!